### ISSUE TITLE
Require Pure Selector Reports False Positive While Nesting

### ISSUE THREAD

**Issue by yuschick** [yuschick/stylelint-plugin-defensive-css#198]
**Describe the bug**
The `require-pure-selectors` rule incorrectly flags nested selectors that extend off of pure parent selectors.

**To Reproduce**
```css
.button {
  color: red;

  &[aria-disabled="true"] {
    color: blue;
  }
}
```

**Expected behavior**
I would expect the nested attribute selector to pass as it extends off of the pure parent class selector.

**Screenshots**

<img width="283" height="187" alt="Image" src="https://github.com/user-attachments/assets/e8c2eb7c-b253-4f28-a648-cefe807edfc2" />

**Additional context**

**Comment by yuschick** [comment]
Resolved in [v2.9.2](https://github.com/yuschick/stylelint-plugin-defensive-css/releases/tag/v2.9.2)
